### PR TITLE
refactor(iroh-automerge): Make the example use the main iroh API instead of iroh-net

### PR DESCRIPTION
**Pull request by matheus23**

### CODE REVIEW

File: iroh-automerge/src/protocol.rs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+    pub fn new(doc: Automerge, sync_finished: mpsc::Sender<Automerge>) -> Arc<Self> {
+        Arc::new(Self {
+            inner: Mutex::new(doc),
+            sync_finished,
+        })
+    }
+
+    pub fn fork_doc(&self) -> Automerge {
+        let automerge = self.inner.lock().expect("lock poisoned");
```
Comment (dignifiedquire) [inline iroh-automerge/src/protocol.rs:42]: use async lock?

**Comment by dignifiedquire** [comment]
very nice, and very happy to see that the protocol extension setup works as expected

File: iroh-automerge/src/main.rs [hunk 2]
```diff
@@ -1,227 +1,83 @@
...
+
     let opts = Cli::parse();
 
-    let mut peer1 = Peer::new().await?;
+    // We set up a channel so we can subscribe to sync events from the automerge protocol
+    let (sync_sender, mut sync_finished) = mpsc::channel(10);
+    let automerge = IrohAutomergeProtocol::new(Automerge::new(), sync_sender);
+    let iroh = Node::memory()
+        .build()
+        .await?
```
Comment (dignifiedquire) [inline iroh-automerge/src/main.rs:33]: should probably disable iroh-docs

**Comment by matheus23** [comment]
I was a little too early with pressing the "ready for review button".
I've expanded the readme & implemented the nits :v:

**Review by dignifiedquire — APPROVED** [review]
very nice

**Pull request merged**